### ISSUE TITLE
cudnn batchnorm layer only works for cnn

### ISSUE THREAD

**Issue by n-gregori** [deeplearning4j/deeplearning4j#3289]
#### Issue Description

batch normalization layer only works with CNN if CUDNN is used.

#### Version Information

* Deeplearning4j 0.8.0
* platform information: centOs (also windows)
* CUDA version: 8.0 

#### Contributing

for example batchnorm with MLP net leads to the following issue:

https://gist.github.com/n-gregori/0138969836c002610ab6eb269dc6a364

**Comment by raver119** [comment]
Issue is fixed in my branch, will be merged soon.
